### PR TITLE
Edit Knobs Feature

### PR DESCRIPTION
**Pull request by arifiorino**
New feature to edit knobs, changes in UI are in red:
Screen Shot 2019-08-07 at 6 14 37 PM
Screen Shot 2019-08-07 at 6 15 17 PM
Screen Shot 2019-08-07 at 6 15 47 PM
Overall changes:
- When you create a session, it makes a new Hardware object with that particular # of CPUs, RAM, and Storage
- A new function set_default_knobs() that sets the min/max values for the knobs when a session is created
- A new edit_knobs page where you can edit individual knobs for that session
- The edited knobs are stored in a new SessionKnob model, and used for the new min/max values.

### CODE REVIEW

**Comment by dvanaken** [comment]
Super excited for this! When it's ready to be reviewed add me as a reviewer.

File: server/website/website/models.py [hunk 1]
```diff
@@ -221,6 +223,34 @@ def delete(self, using=DEFAULT_DB_ALIAS, keep_parents=False):
         super(Session, self).delete(using=DEFAULT_DB_ALIAS, keep_parents=False)
 
 
+class SessionKnobManager(models.Manager):
+    @staticmethod
+    def get_knobs_for_session(session):
+            # Returns a dict of the knob
+        knobs = KnobCatalog.objects.filter(dbms=session.dbms)
```
Comment (eribeiro) [inline server/website/website/models.py]: This suggestion goes from line 230 up to 239. If you eventually decide it's worth incorporating then it still needs to be double checked to see if it is a perfect replacement to your original code section.

```python
   knobs = list(KnobCatalog.objects.filter(dbms=session.dbms))
   session_knobs = SessionKnob.objects.filter(session=session, knob__in=knobs)
   for sk in session_knobs:
        i = knobs.index(sk.knob)
        knobs[i].minval = sk.minval 
        knobs[i].maxval = sk.maxval
        knobs[i].tunable = sk.tunable

   return [k for k in knobs.values() if k["tunable"]]

**Comment by dvanaken** [comment]
@arifiorino can you update your branch so I can merge?

**Comment by arifiorino** [comment]
@dvanaken When I try to merge the ottertune master into this branch it says "Already up to date" am I doing something wrong? I also tried "git rebase" but I don't think it changed anything.

**Comment by dvanaken** [comment]
Not sure why it wouldn't let me rebase & merge since your branch was up to date but I was able to do a normal merge.

**Pull request merged**